### PR TITLE
Add openapi yaml data loader

### PR DESCRIPTION
**Pull request by deven298**
## Description

This PR adds support for loading OpenAPI's yaml config file.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (does not change functionality, e.g. code style improvements, linting)
- [ ] Documentation update

## How Has This Been Tested?

I tested this by writing Unit Test and also by writing the Test Script.

Please delete options that are not relevant.

- [x] Unit Test
- [x] Test Script (please provide)

Test Script
```
from embedchain.apps.app import App
import os

os.environ["OPENAI_API_KEY"] = "openai_api_key"

app = App()

response = app.query("What can OpenAI API endpoint do? Can you list the things it can learn from?")

print("Respose: ", response)

app.add("configs/openai_openapi.yaml")

print("Successfully added openapi yaml content!")

response = app.query("What can OpenAI API endpoint do? Can you list the things it can learn from?")

print("Respose: ", response)
```

YAML file
[https://github.com/openai/openai-openapi/blob/master/openapi.yaml](url)

Output: Learned where can user find official documentation of the OpenAI API from its OpenAPI yaml config file.
```
Respose:  The OpenAI API endpoint is a powerful tool that allows developers to integrate OpenAI's language models into their applications. It can learn from a wide range of text sources, including books, articles, websites, and other written content. The API endpoint can be used for various tasks such as generating text, answering questions, summarizing documents, translating languages, and much more.
Successfully saved configs/openai_openapi.yaml (DataType.OPENAPI). New chunks count: 6
Successfully added openapi yaml content!
Respose:  The OpenAI API endpoint allows users to interact with OpenAI's models and perform various tasks such as generating text, answering questions, summarizing documents, translating languages, and more. The specific capabilities and tasks that the API can learn from may vary depending on the models and features provided by OpenAI. For more detailed information, it is recommended to refer to the OpenAI API documentation at https://platform.openai.com/docs/api-reference.
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

File: embedchain/utils.py [hunk 1]
```diff
@@ -155,6 +155,10 @@ def detect_datatype(source: Any) -> DataType:
             logging.debug(f"Source of `{formatted_source}` detected as `docx`.")
             return DataType.DOCX
 
+        if url.path.endswith(".yaml"):
+            logging.debug(f"Source of `{formatted_source}` detected as `yaml`.")
+            return DataType.OPENAPI
```
Comment (deshraj) [inline embedchain/utils.py:160]: I am not sure if we want to detect all the `*.yaml` files as openapi data type.

**Review by deshraj — APPROVED** [review]
LGTM!! Left a minor comment. 

Also, can you please update the docs?

File: embedchain/utils.py (hunk 1 above)
Comment (deven298) [reply embedchain/utils.py:160]: True, adding the safeguard for openapi type yaml file would be a good idea.

**Comment by deshraj** [comment]
@deven298 can you please fix the lint issues listed here for this PR: https://github.com/embedchain/embedchain/actions/runs/6579064752/job/17874138608?pr=818

Once these are resolved, we are ready to merge it. Thanks!

File: embedchain/chunkers/openapi.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+
+from embedchain.chunkers.base_chunker import BaseChunker
+from embedchain.config.add_config import ChunkerConfig
+
+
+class OpenAPIChunker(BaseChunker):
+    def __init__(self, config: ChunkerConfig | None = None):
```
Comment (deshraj) [inline embedchain/chunkers/openapi.py:8]: This is why the tests are failing I think.

Comment (deven298) [reply embedchain/chunkers/openapi.py:8]: Hmm. But it works locally, thats weird.

**Comment by deven298** [comment]
Resolved all conflicts and ready for merge.

**Pull request merged**